packages/gestalt/src/PageHeaderSubcomponents.js

```diff
@@ -0,0 +1,200 @@
+        smDisplay="block"
+        dangerouslySetInlineStyle={{ __style: { marginTop: '-7px' } }}
+      >
+        <Heading size="500" lineClamp={1} accessibilityLevel={1}>
+          {title}
+        </Heading>
+      </Box>
+    </Fragment>
+  );
+}
+
+export function PageHeaderThubnail({ thumbnail }: {| thumbnail?: Element<typeof Image> |}): Node {
+  return (
+    <Box display="none" smDisplay="block">
+      <Mask height={48} rounding={2} width={48}>
+        {thumbnail}
+      </Mask>
+    </Box>
+  );
+}
+
+export function PageHeaderBadge({
+  badgeText,
+  badgeTooltipText,
+}: {|
+  badgeText: string,
+  badgeTooltipText?: string,
+|}): Node {
+  return (
+    <Fragment>
+      {badgeText && !badgeTooltipText ? (
+        <Badge text={badgeText} type="info" position="top" />
+      ) : null}
+      {badgeText && badgeTooltipText ? (
+        <Tooltip accessibilityLabel="" text={badgeTooltipText} idealDirection="up">
+          <Badge text={badgeText} type="info" position="top" />
+        </Tooltip>
+      ) : null}
+    </Fragment>
+  );
+}
+
+export function PageHeaderHelperIconButton({
+  helperIconButton,
+}: {|
+  helperIconButton?: {|
+    accessibilityLabel: string,
+    accessibilityControls: string,
+    accessibilityExpanded: boolean,
+    onClick: ({|
+      event:
+        | SyntheticMouseEvent<HTMLButtonElement>
+        | SyntheticKeyboardEvent<HTMLButtonElement>
+        | SyntheticMouseEvent<HTMLAnchorElement>
+        | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+  |},
+|}): Node {
+  return helperIconButton ? (
+    <IconButton
+      accessibilityControls={helperIconButton.accessibilityControls}
+      accessibilityExpanded={helperIconButton.accessibilityExpanded}
+      accessibilityHaspopup
+      accessibilityLabel={helperIconButton.accessibilityLabel}
+      bgColor="lightGray"
+      icon="question-mark"
+      iconColor="darkGray"
+      onClick={helperIconButton.onClick}
+      size="xs"
+    />
+  ) : null;
+}
+
+export function PageHeaderSubtext({
+  subtext,
+  helperLink,
+}: {|
+  subtext?: string,
+  helperLink?: {|
+    text: string,
+    accessibilityLabel: string,
+    href: string,
+    onClick: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+  |},
+|}): Node {
+  return (
+    <Box wrap display="none" smDisplay="block">
+      <Text inline>
+        {subtext}{' '}
+        {helperLink ? (
+          <Link
+            inline
+            accessibilityLabel={helperLink.accessibilityLabel}
+            href={helperLink.href}
+            onClick={helperLink.onClick}
+            target="blank"
+          >
+            <span style={{ textDecoration: 'underline' }}>{helperLink.text}</span>
+          </Link>
+        ) : null}
+      </Text>
+    </Box>
+  );
+}
+
+export function PageHeaderActionBlock({
```

Comment:
This API is overly permissive and will allow devs to unwittingly create bad UIs. We need to guard against several scenarios:
- Only secondary action with no primary action (I could be wrong, we may want to allow this)
- Only actions with no dropdown items
- Only dropdown items with no actions
- Mismatch between actions and dropdown items
- No label provided

We should check for these scenarios and throw errors if needed.

I wonder if tighter coupling of actions and dropdown items in the API would make that connection better. Perhaps each action takes both a button and dropdown item in the same prop, to ensure that devs provide both (two required fields on the value of the optional prop)

Reply:
@dangerismycat 

Decided to go the error route. 

"Mismatch between actions and dropdown items" it's hard cos Action can be IconButton with Dropdown and several items there. So it isn't one to one situation.